src/main/java/com/farao_community/farao/gridcapa/data_bridge/sources/SftpSource.java

```diff
@@ -6,97 +6,122 @@
+    public void allSFTP() throws IOException {
+        for (DataBridge bridge : remoteFilesConfiguration.getDataBridgeList()) {
+
+            MessageChannel channel = new PublishSubscribeChannel();
+            this.applicationContext.getAutowireCapableBeanFactory().initializeBean(channel, bridge.getBridgeIdentifiant() + "_sftp");
+
+            IntegrationFlow flow = IntegrationFlows.from(bridge.getBridgeIdentifiant() + "_flow")
+                    .channel("archivesChannel")
+                    .get();
+
+            IntegrationFlow ms = this.sftpInboundFlow(bridge);
+            this.applicationContext.getAutowireCapableBeanFactory().initializeBean(ms, bridge.getBridgeIdentifiant() + "_sftp_adapter");
 
-    @Bean
-    @InboundChannelAdapter(channel = "sftpSourceChannel", poller = @Poller(fixedDelay = "${data-bridge.sources.sftp.polling-delay-in-ms}"))
-    public MessageSource<File> sftpMessageSource() throws IOException {
-        SftpInboundFileSynchronizingMessageSource source = new SftpInboundFileSynchronizingMessageSource(sftpInboundFileSynchronizer());
-        source.setLocalDirectory(Files.createTempDirectory(SYNCHRONIZE_TEMP_DIRECTORY_PREFIX).toFile());
-        source.setAutoCreateLocalDirectory(true);
-        source.setLocalFilter(new FileSystemPersistentAcceptOnceFileListFilter(new SimpleMetadataStore(), ""));
-        return source;
+            this.applicationContext.getAutowireCapableBeanFactory().initializeBean(flow, bridge.getBridgeIdentifiant() + "_sftp_flow");
+
+        }
     }
 
-    @Bean
-    public IntegrationFlow sftpPreprocessFlow() {
-        return IntegrationFlows.from("sftpSourceChannel")
-                .channel("archivesChannel")
+    public IntegrationFlow sftpInboundFlow(DataBridge bridge) throws IOException {
+        CompositeFileListFilter<ChannelSftp.LsEntry> fileListFilter = new CompositeFileListFilter<>();
+        fileListFilter.addFilter(new SftpPersistentAcceptOnceFileListFilter(new SimpleMetadataStore(), ""));
+        fileListFilter.addFilter(new SftpRegexPatternFileListFilter(String.join("|", bridge.getRemoteFileRegex())));
+        return IntegrationFlows
+                .from(Sftp.inboundAdapter(sftpSessionFactory())
+                                .preserveTimestamp(true)
+                                .deleteRemoteFiles(false)
+                                .remoteDirectory(bridge.getFtpDirectory())
+                                .filter(fileListFilter)
+                                .autoCreateLocalDirectory(true)
+                                .localFilter(new FileSystemPersistentAcceptOnceFileListFilter(new SimpleMetadataStore(), ""))
+                                .localDirectory(Files.createTempDirectory(SYNCHRONIZE_TEMP_DIRECTORY_PREFIX).toFile()),
+                    e -> e.id(bridge.getBridgeIdentifiant() + "_sftp")
+                                .autoStartup(true)
+                                .poller(Pollers.fixedDelay(polling))
+
+                )
```

Comment:
Can be refactored a bit to be more readable

Reply:
refactor in the same way as FtpSource